framework/derive/src/generate/proxy_gen.rs

```diff
@@ -298,6 +298,24 @@ pub fn proxy_trait(contract: &ContractTrait) -> proc_macro2::TokenStream {
     }
 }
 
+pub fn tx_proxy_trait() -> proc_macro2::TokenStream {
+    quote! {
+        pub trait TxProxyTrait<Env>{
```

Comment:
There is only one `TxProxyTrait` and one `TxProxyMethods`, they don't need to be generated for each contract. You can place them directly somewhere in types/interaction, in framework base.

The derive crate remains untouched in thie PR.